presto-main/src/test/java/com/facebook/presto/connector/system/TestSystemSession.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.connector.system;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.common.RuntimeStats;
+import com.facebook.presto.common.function.SqlFunctionProperties;
+import com.facebook.presto.common.transaction.TransactionId;
+import com.facebook.presto.common.type.TimeZoneKey;
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.QueryId;
+import com.facebook.presto.spi.WarningCollector;
+import com.facebook.presto.spi.connector.ConnectorTransactionHandle;
+import com.facebook.presto.spi.function.SqlFunctionId;
+import com.facebook.presto.spi.function.SqlInvokedFunction;
+import com.facebook.presto.spi.security.ConnectorIdentity;
+import com.facebook.presto.spi.security.SelectedRole;
+import org.testng.annotations.Test;
+
+import java.security.Principal;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+
+public class TestSystemSession
+{
+    @Test
+    public void testSystemSessionForExtraCredentials()
+    {
+        TransactionId transactionId = new TransactionId(UUID.randomUUID());
+
+        ConnectorTransactionHandle transactionHandle = new GlobalSystemTransactionHandle("connector-id", transactionId);
+
+        Optional<Principal> principal = Optional.of(() -> "principal");
+
+        Optional<SelectedRole> selectedRole = Optional.empty();
+
+        ConnectorIdentity connectorIdentity = new ConnectorIdentity("user", principal, selectedRole) {
+            @Override
+            public Map<String, String> getExtraCredentials()
+            {
+                Map<String, String> extraCredentials = new HashMap<>();
+                extraCredentials.put("key1", "value1");
+                extraCredentials.put("key2", "value2");
+                return extraCredentials;
+            }
+        };
+
+        String validQueryId = "query_id_1";
+        QueryId queryId = new QueryId(validQueryId);
+
+        ConnectorSession session = new ConnectorSession() {
```

Comment:
Please just use a TestingConnectorSession.

Reply:
@tdcmeehan Used TestingConnectorSession. Could you have  a look?
Thanks.